From af0991b2d7cd77fc077b31d4d67a556bcbd235fe Mon Sep 17 00:00:00 2001
Message-Id: <af0991b2d7cd77fc077b31d4d67a556bcbd235fe.1576111643.git.liu.denton@gmail.com>
In-Reply-To: <cover.1576111643.git.liu.denton@gmail.com>
References: <cover.1576111643.git.liu.denton@gmail.com>
From: Denton Liu <liu.denton@gmail.com>
Date: Wed, 11 Dec 2019 15:30:20 -0800
Subject: [PATCH 1/2] config/format.txt: clarify behavior of multiple
 format.notes
To: Git Mailing List <git@vger.kernel.org>
Cc: Elijah Newren <newren@gmail.com>,
    Eric Sunshine <sunshine@sunshineco.com>,
    Philip Oakley <philipoakley@iee.email>

In 8164c961e1 (format-patch: use --notes behavior for format.notes,
2019-12-09), we slightly tweaked the behavior of having multiple
`format.notes` configuration variables. We did not update the
documentation to reflect this, however.

Explictly state the behavior of having multiple `format.notes`
configuration variables so users are clear on what to expect.

Signed-off-by: Denton Liu <liu.denton@gmail.com>
---
 Documentation/config/format.txt | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/Documentation/config/format.txt b/Documentation/config/format.txt
index 414a5a8a9d..3d708b0aec 100644
--- a/Documentation/config/format.txt
+++ b/Documentation/config/format.txt
@@ -99,4 +99,20 @@ If one wishes to use the ref `ref/notes/true`, please use that literal
 instead.
 +
 This configuration can be specified multiple times in order to allow
-multiple notes refs to be included.
+multiple notes refs to be included. In that case, it will behave
+similarly to multiple `--[no-]notes[=]` options passed in. That is, a
+value of `true` will show the default notes, a value of `<ref>` will
+also show notes from that notes ref and a value of `false` will negate
+previous configurations and not show notes.
++
+For example,
++
+------------
+[format]
+	notes = true
+	notes = foo
+	notes = false
+	notes = bar
+------------
++
+will only show notes from `refs/notes/bar`.
-- 
2.24.0.627.geba02921db

